drivers/spi/spi_mcux_dspi.c

```diff
@@ -282,27 +773,112 @@ static const struct spi_driver_api spi_mcux_driver_api = {
+		.dma_dev =						\
+		    DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(id, tx)),	\
+		.dma_cfg = {						\
+			.source_burst_length = 4,			\
+			.dest_burst_length = 4,				\
+			.source_data_size = 4,				\
+			.dest_data_size = 4,				\
+			.dma_callback = dma_callback,			\
+			.complete_callback_en = 1,			\
+			.error_callback_en = 1,				\
+			.block_count = 1,				\
+			.head_block = &spi_mcux_data_##id.tx_dma_block,	\
+			.channel_direction = MEMORY_TO_PERIPHERAL,	\
+			.dma_slot = DT_INST_DMAS_CELL_BY_NAME(		\
+				id, tx, source),			\
+		},							\
+	},
+
+#define RX_DMA_CONFIG(id)						\
+	.inner_rx_buffer = &spi_edma_rx_buffer_##id,			\
+	.rx_dma_config = {						\
+		.dma_dev =						\
+		    DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(id, rx)),	\
+		.dma_cfg = {						\
+			.source_burst_length = 2,			\
+			.dest_burst_length = 2,				\
+			.source_data_size = 2,				\
+			.dest_data_size = 2,				\
+			.dma_callback = dma_callback,			\
+			.complete_callback_en = 1,			\
+			.error_callback_en = 1,				\
+			.block_count =					\
+			_UTIL_AND2(DT_INST_NODE_HAS_PROP(		\
+				id, nxp_rx_tx_chn_share), 2),		\
+			.head_block = &spi_mcux_data_##id.rx_dma_block,	\
+			.channel_direction = PERIPHERAL_TO_MEMORY,	\
+			.dma_slot = DT_INST_DMAS_CELL_BY_NAME(		\
+				id, rx, source),			\
+		},							\
+	},
+#else
+#define TX_BUFFER(id)
+#define RX_BUFFER(id)
+#define TX_DMA_CONFIG(id)
+#define RX_DMA_CONFIG(id)
+
+#endif
+
 #define SPI_MCUX_DSPI_DEVICE(id)					\
-	static void spi_mcux_config_func_##id(const struct device *dev); \
+	static void spi_mcux_config_func_##id(const struct device *dev);\
+	TX_BUFFER(id);							\
+	RX_BUFFER(id);							\
+	static struct spi_mcux_data spi_mcux_data_##id = {		\
+		SPI_CONTEXT_INIT_LOCK(spi_mcux_data_##id, ctx),		\
+		SPI_CONTEXT_INIT_SYNC(spi_mcux_data_##id, ctx),		\
+		TX_DMA_CONFIG(id) RX_DMA_CONFIG(id)			\
+	};								\
 	static const struct spi_mcux_config spi_mcux_config_##id = {	\
 		.base = (SPI_Type *)DT_INST_REG_ADDR(id),		\
 		.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(id)),	\
 		.clock_subsys = 					\
 		(clock_control_subsys_t)DT_INST_CLOCKS_CELL(id, name),	\
 		.irq_config_func = spi_mcux_config_func_##id,		\
-		.pcs_sck_delay = UTIL_AND(				\
-			DT_INST_NODE_HAS_PROP(id, pcs_sck_delay),	\
-			DT_INST_PROP(id, pcs_sck_delay)),		\
-		.sck_pcs_delay = UTIL_AND(				\
-			DT_INST_NODE_HAS_PROP(id, sck_pcs_delay),	\
-			DT_INST_PROP(id, sck_pcs_delay)),		\
-		.transfer_delay = UTIL_AND(				\
-			DT_INST_NODE_HAS_PROP(id, transfer_delay),	\
-			DT_INST_PROP(id, transfer_delay)),		\
-	};								\
-	static struct spi_mcux_data spi_mcux_data_##id = {		\
-		SPI_CONTEXT_INIT_LOCK(spi_mcux_data_##id, ctx),		\
-		SPI_CONTEXT_INIT_SYNC(spi_mcux_data_##id, ctx),		\
+		.pcs_sck_delay =					\
+		    DT_INST_PROP_OR(id, pcs_sck_delay, 1000),		\
+		.sck_pcs_delay =					\
+		    DT_INST_PROP_OR(id, sck_pcs_delay, 1000),		\
+		.transfer_delay =					\
+		    DT_INST_PROP_OR(id, transfer_delay, 1000),		\
```

Comment:
Why are you changing the defaults away from zero?

Reply:
this is copied from getDefault setting of SDK.